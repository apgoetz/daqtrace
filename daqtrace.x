SECTIONS
{
  /* `0` specifies the start address of this virtual (`(INFO)`) section */
  .daqtrace 0 (INFO) :
  {

    /* Everything user-defined */
    *(.daqtrace);
  }
}

ASSERT(SIZEOF(.daqtrace) < (1 << 16), ".daqtrace section id must fit in u16");
