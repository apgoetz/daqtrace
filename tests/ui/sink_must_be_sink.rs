use daqtrace::trace;

fn main() {
    let sink = 1;
    let (a,b,c) = (1,2,3);
    trace!(sink, hello, a, b, c);
}

