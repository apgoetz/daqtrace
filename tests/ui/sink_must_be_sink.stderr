error[E0277]: the trait bound `{integer}: daqtrace::Sink` is not satisfied
 --> $DIR/sink_must_be_sink.rs:6:12
  |
6 |     trace!(sink, hello, a, b, c);
  |            ^^^^ the trait `daqtrace::Sink` is not implemented for `{integer}`
  |
  = note: required by `timestamp`
